phare/pp/diagnostics.py

```diff
@@ -0,0 +1,234 @@
+# formatted with black code formatter
+
+import os, h5py, phare.pharein as ph, itertools
+from .diagnostic.patch_data import _EM, _Fluid, _Particle, Particles
+from .diagnostic.patch import Patch, PatchLevel
+
+
+class H5FileRegexer:
+    """Used to Validates h5 input file to ignore files which are not to be read/extracted"""
+
+    @staticmethod
+    def ignore_population_name_in(h5FilesPerType):
+        import re
+
+        pop = "${POP}"
+        regexing = {Type: [] for Type in h5FilesPerType.keys()}
+        for Type, files in h5FilesPerType.items():
+            for file in files:
+                if pop not in file:
+                    continue
+                bits = file.split(pop)
+                left, right = bits[0], bits[1]
+                regexing[Type].append(
+                    (file, re.compile("^" + left + "(.*)" + right + "$"))
+                )
+        return regexing
+
+
+class Diagnostic:
+    REFINEMENT_RATIO = 2
+    """12 decimal places is arbitrary, but hopefully sufficient"""
+    DL_PRECISION = 12
+
+    """List of all possible input HDF5 Files"""
+    h5FilesPerType = {
+        _EM: ["EM_B.h5", "EM_E.h5"],
+        _Fluid: [
+            "ions_bulkVelocity.h5",
+            "ions_density.h5",
+            "ions_pop_ions_${POP}_density.h5",
+            "ions_pop_ions_${POP}_flux.h5",
+        ],
+        _Particle: [
+            "ions_pop_ions_${POP}_domain.h5",
+            "ions_pop_ions_${POP}_patchGhost.h5",
+            "ions_pop_ions_${POP}_levelGhost.h5",
+        ],
+    }
+    h5Files = [f for files in itertools.chain(h5FilesPerType.values()) for f in files]
+
+    """Compiled Regexs to ignore population name in HDF5 input files"""
+    h5FilesRegexesPerType = H5FileRegexer.ignore_population_name_in(h5FilesPerType)
+    h5FilesRegexes = {
```

Comment:
hard for me to know what that dict looks like in the end... can you explain

Reply:
done